components/server/src/user/user-deletion-service.ts

```diff
@@ -140,6 +145,23 @@ export class UserDeletionService {
         await Promise.all(teams.map(t => this.teamDb.removeMemberFromTeam(userId, t.id)));
     }
 
+    protected async deleteTeams(userId: string) {
+        const ownedTeams = await this.teamDb.findTeamsByUserAsSoleOwner(userId);
+
+        ownedTeams.forEach(async team => {
+            const teamProjects = this.projectDb.findTeamProjects(team.id);
+            (await teamProjects).forEach(project => this.projectDb.markDeleted(project.id));
+        })
```

Comment:
Minor suggestion: Maybe it's worth waiting for all projects to be removed before deleting the teams? I believe the `forEach` will just trigger multiple parallel promises on-the-fly without `await`ing them. If we wanted to wait for completion on these calls, I believe this could be rewritten like so:

```suggestion
        for (const team of ownedTeams) {
            const teamProjects = await this.projectDb.findTeamProjects(team.id);
            await Promise.all(teamProjects.map(project => this.projectDb.markDeleted(project.id)));
        }
```

Reply:
You're absolutely right 🙏🏽